components/domain/search-result-item.tsx

```
@@ -0,0 +1,150 @@
+      </div>
+    </div>
+  )
+}
+
+// Keyboard navigation helper for lists of search results
+export function useSearchResultNavigation(results: SearchResult[], onSelect: (result: SearchResult) => void) {
+  const handleKeyDown = (e: React.KeyboardEvent, currentIndex: number) => {
+    switch (e.key) {
+      case "ArrowDown":
+        e.preventDefault()
+        const nextIndex = Math.min(currentIndex + 1, results.length - 1)
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)